cluster-autoscaler/cloudprovider/gce/mig_instance_templates_provider.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"sync"
+	"time"
+
+	gce "google.golang.org/api/compute/v1"
+)
+
+const (
+	migInstanceCacheRefreshInterval = 60 * time.Minute
```

Comment:
1. Having those timeouts scattered all over the place makes them hard to find and quickly figure out what is cached and for how long. Maybe we should just put them all in one place, to make it easier? Something like cloudprovider/gce/timeouts.go? WDYT?
2. I'm trying to figure out if those can ever change? Upgrading to a newer k8s version is the only thing that comes to my mind. Are there other scenarios? Even if there aren't 1 hour seems like a long timeout. Maybe we should make it 10 minutes or so?
3. On a somewhat related question (not part of this PR, just something to consider for future) - we're going to get one much slower loop every time we drop cache. Maybe we should consider including some random jitter to timeout, so that we invalidate instances for different groups at different times and spread the pain between loops? Same goes for other caching we do in cloudprovider.

Reply:
(3) is the problem I also though about and I do not like the behavior. The anychronous approach would let as avoid this problem. 
Having `lastRefresh` per each entry in the map would help too - but is kinda ugly and not sure who should hold it - cache itself or the provider. I will give it a though and maybe send out a followup.

Reply:
> Having those timeouts scattered all over the place makes them hard to find and quickly figure out what is cached and for how long. Maybe we should just put them all in one place, to make it easier? Something like cloudprovider/gce/timeouts.go? WDYT

Actually other timeouts are outside of GCE cloud provider. And are part of core logic - so there is no natural place group them. I will leave the const here and only start grouping if we end up having more of those.

Reply:
(2) "szewskim targiem" changing to 30s